### ISSUE TITLE
NuGet pack doesn't work without specifying nuspec or project file

### ISSUE THREAD

**Issue by brgrz** [NuGet/Home#1985]
Nuget pack used to work in 2.5 and 2.8 but after upgrading nuget.exe to 3.3 it now outputs this:

MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'.
Please specify a nuspec or project file to use.

It should pick up the nuspec in the current directory. This change wasn't documented in release notes here https://docs.nuget.org/release-notes/nuget-3.3

**Comment by yishaigalatzer** [comment]
@csharpfritz would you mind documenting this for now. This seems like a good catch, but a low pri. We'd take a fix if someone wants to contribute one

**Comment by brgrz** [comment]
How's this low priority? Pack is core feature. I tried to run it with specifying the .nuspec file and it resulted in https://github.com/NuGet/Home/issues/1986

I did no code changes whatsoever just upgraded nuget.exe from 2.8.6 to 3.3.0.212

**Comment by yishaigalatzer** [comment]
Low priority because you can specify it, and you will be unblocked.

**Comment by toddm** [comment]
I just fixed this - it was incorrectly showing the error if there was a nuspec and a project file in the directory.  Now it defaults to packing using the project file (which uses the nuspec as well).

**Comment by jibeiz** [comment]
Just so you know, this issue happens again to me with nuget.exe (v3.4.4)

**Comment by rohit21agrawal** [comment]
@jibeiz what kind of project are you using? I just tried nuget-3.4.4 and nuget-3.5.0-rc with a simple console application and ran nuget pack  without specifying a csproj or nuspec file and it packed as expected.

**Comment by jibeiz** [comment]
@rohit21agrawal i was trying to pack a C# project, using the nuget Commandline downloaded from [here ](https://dist.nuget.org/index.html). There is a .nuspec and a .csproj in the direcotry at the sametime. I tried to run "nuget pack" with v3.3.0 and v3.4.4, they gave me "Please specify a nuspec or project file to use", then i tried v2.8.6, which worked fine. thx

**Comment by rohit21agrawal** [comment]
@jibeiz is it possible for you to attach a sample project which repros the problem?

**Comment by yishaigalatzer** [comment]
And the exact commandline you run

On Aug 24, 2016 7:46 AM, Rohit Agrawal notifications@github.com wrote:

@jibeizhttps://github.com/jibeiz is it possible for you to attach a sample project which repros the problem?

##

**Comment by falahati** [comment]
This is still the case. Using following command line in a target file results in the fallowing message:

>   NuGet.exe pack -Verbosity detailed -Symbols -OutputDir "NuGet" -Version 1.0.0.0 -Properties "Configuration=Release"
>   NuGet.CommandLineException: Please specify a nuspec or project file to use.
>      at NuGet.CommandLine.PackCommand.GetInputFile(IEnumerable`1 files)
>      at NuGet.CommandLine.PackCommand.GetInputFile()
>      at NuGet.CommandLine.PackCommand.ExecuteCommand()
>      at NuGet.CommandLine.Command.ExecuteCommandAsync()
>      at NuGet.CommandLine.Command.Execute()
>      at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)
> 
> ....\NuGet.targets(12,5): error MSB3073: The command "NuGet.exe pack -Verbosity detailed -Symbols -OutputDir "NuGet" -Version 1.0.0.0 -Properties "Configuration=Release"" exited with code 1. [...\MyProject.csproj]
> 
> Done Building Project "...\MyProject.csproj" (Build;Package target(s)) -- FAILED.

My target file is:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Target Name="AfterBuild">
    <GetAssemblyIdentity AssemblyFiles="$(OutputPath)$(AssemblyName).dll">
      <Output TaskParameter="Assemblies" ItemName="OutputAssemblyInfo" />
    </GetAssemblyIdentity>
    <Message Text="Info: %(OutputAssemblyInfo.Version)" />
  </Target>
  <Target Name="Package">
    <RemoveDir Directories="NuGet" />
    <MakeDir Directories="NuGet" />
    <Exec WorkingDirectory="$(BuildDir)" Command="NuGet.exe pack -Verbosity detailed -Symbols -OutputDir &quot;NuGet&quot; -Version %(OutputAssemblyInfo.Version) -Properties &quot;Configuration=$(Configuration)&quot;" />
  </Target>
  <Target Name="Publish">
    <Exec WorkingDirectory="$(BuildDir)" Command="NuGet.exe push &quot;NuGet\$(AssemblyName).%(OutputAssemblyInfo.Version).nupkg&quot; -Verbosity detailed" />
  </Target>
</Project>
```

My build.ps is:

``` ps1
$msbuild = join-path -path (Get-ItemProperty "HKLM:\software\Microsoft\MSBuild\ToolsVersions\14.0")."MSBuildToolsPath" -childpath "msbuild.exe"
&$msbuild MyProject\MyProject.csproj /t:Build /t:Package /p:Configuration="Release"
```

Works well with 2.8 version, but not 3.x; but then 2.8 support for PCL is not very good

**Comment by yishaigalatzer** [comment]
Does it work from the command line? What version are you using?

**Comment by falahati** [comment]
Not until I explicitly mention the project file I want:

> NuGet.exe pack -Verbosity detailed -Symbols -OutputDir "NuGet2" -Version 1.0.0.0 -Properties "Configuration=$(Configuration)"
> Fails, while
> 
> NuGet.exe pack "MyProject.csproj" -Verbosity detailed -Symbols -OutputDir "NuGet2" -Version 1.0.0.0 -Properties "Configuration=$(Configuration)"

Works. Not a big issue, but a change of behavior in current major version comparing to last major release. Version 2.8 would have detected the project file if started from project directory.

**Comment by yishaigalatzer** [comment]
What version are you using? Can you try downloading 3.5 from myget/nugetbuild? Build # is 1938

**Comment by brgrz** [comment]
@yishaigalatzer What if for once you tried to repro the issue instead of constantly asking what versions we're using? If you don't understand the issue (which you clearly don't coz you keep asking the same question all over again for the last 10 months), haven't reproduced it then you most likely didn't fix it and telling us to use the new version won't help. It will just prove it still doesn't work. I'm willing to put a bet on it, are you?

We're always using the latest or almost latest version.

Seriously, this is unbelievable. FFS just put a .csproj and .nuspec file in the same folder, run nuget pack and you'll have your repro.

**Comment by yishaigalatzer** [comment]
Brgrz read the thread, the engineer said he fixed it, and also said he cannot reproduce it locally.

Providing a report without details doesn't help anyone, and I'm currently not on a computer and can't reproduce it from a phone ...

**Comment by falahati** [comment]
I was on 3.4.4, lastest version
tried with 3.6 nightly build 1984, works as expected

> NuGet.exe pack -Verbosity detailed -Symbols -OutputDir "NuGet2" -Version 1.0.0.0 -Properties "Configuration=$(Configuration)"
> NuGet Version: 3.6.0.1984
> Attempting to build package from 'MyProject.csproj'.

we are safe to leave this issue alone then xD

**Comment by yishaigalatzer** [comment]
Great, thanks so much for validating.

**Comment by falahati** [comment]
Yaeh, well but there is an other problem.
If executed from target file, nuget fails in multiple ways:
- It cant find the csproj file (3.6)
- But when csproj file address is added to the target file, then it cant find the nuspec file next to csproj file (3.6)

Second problem is not in 3.4 version. Currently I am going with 3.4 version because the second problem is a lot more important. However, 3.4 version cant find the dependencies correctly.

you guys need to do something about this inconsistency between versions.
